File: java/domain/src/test/java/dev/ufadevconf/domain/PickupPointTest.java
package dev.ufadevconf.domain;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class PickupPointTest {

    @Test
    @DisplayName("Пункт выдачи принимает посылку если вес в допустимых границах")
    void acceptsParcelWithinRange() {
        var pickupPoint = new PickupPoint(
                "SPB-101",
                "Санкт-Петербург, Невский проспект, 1",
                new Weight(1000),
                new Weight(5000)
        );

        var lightParcel = new Parcel("TRACK-1", new Weight(1500), "Документы");
        var heavyParcel = new Parcel("TRACK-2", new Weight(7000), "Инструменты");

        assertThat(pickupPoint.canAccept(lightParcel)).isTrue();
        assertThat(pickupPoint.canAccept(heavyParcel)).isFalse();
    }

    @Test
    @DisplayName("Пункт выдачи отклоняет посылку легче минимального порога")
    void rejectsParcelBelowMinimum() {
        var pickupPoint = new PickupPoint(
                "SPB-102",
                "Санкт-Петербург, Литейный проспект, 5",
                new Weight(2000),
                new Weight(6000)
        );

        var tooLightParcel = new Parcel("TRACK-3", new Weight(1500), "Образцы");

        assertThat(pickupPoint.canAccept(tooLightParcel)).isFalse();
    }
}

File: kotlin/domain/src/test/kotlin/dev/ufadevconf/domain/PickupPointTest.kt
package dev.ufadevconf.domain

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

class PickupPointTest {

    @Test
    @DisplayName("Пункт выдачи принимает посылку если вес в допустимых границах")
    fun acceptsParcelWithinRange() {
        val pickupPoint = PickupPoint(
            code = "SPB-101",
            address = "Санкт-Петербург, Невский проспект, 1",
            minWeight = Weight(1000),
            maxWeight = Weight(5000)
        )

        val lightParcel = Parcel("TRACK-1", Weight(1500), "Документы")
        val heavyParcel = Parcel("TRACK-2", Weight(7000), "Инструменты")

        assertThat(pickupPoint.canAccept(lightParcel)).isTrue()
        assertThat(pickupPoint.canAccept(heavyParcel)).isFalse()
    }

    @Test
    @DisplayName("Пункт выдачи отклоняет посылку легче минимального порога")
    fun rejectsParcelBelowMinimum() {
        val pickupPoint = PickupPoint(
            code = "SPB-102",
            address = "Санкт-Петербург, Литейный пр., 5",
            minWeight = Weight(2000),
            maxWeight = Weight(6000)
        )

        val tooLightParcel = Parcel("TRACK-3", Weight(1500), "Образцы")

        assertThat(pickupPoint.canAccept(tooLightParcel)).isFalse()
    }
}
